src/prefect/tasks/cubejs/cubejs_utils.py

```diff
@@ -0,0 +1,179 @@
+import jwt
+from requests import Session
+import time
+from typing import Dict, Union
+
+from prefect.engine.signals import FAIL
+
+
+class CubeJSClient:
+
+    # Cube Cloud base URL
+    __CUBEJS_CLOUD_BASE_URL = "https://{subdomain}.cubecloud.dev"
+
+    def __init__(
+        self,
+        subdomain: str,
+        url: str,
+        security_context: Union[str, Dict],
+        secret: str,
+        wait_api_call_secs: int,
+        max_wait_time: int,
+    ):
+        """
+        Initialize a `CubeJSClient`.
+        The client can be used to interact with Cube.js APIs.
+
+        Args:
+            - subdomain (str): Cube Cloud subdomain.
+            - url (str): Cube.js URL (likely to be used in self-hosted Cube.js
+                deployments).
+            - security_context (str, dict): The security context to be used
+                when interacting with Cube.js APIs.
+            - secret (str): The secret string to be used, together with the
+                `security_context`, to generate the API token to pass in the
+                authorization header.
+            - wait_api_call_secs (int): Number of seconds to wait
+                between API calls.
+            - max_wait_time (int): The maximum amount of seconds to wait for
+                an API call to respond.
+        """
+        self.cube_base_url = self.get_cube_base_url(subdomain=subdomain, url=url)
+        self.api_token = self.get_api_token(
+            security_context=security_context, secret=secret
+        )
+        self.wait_api_call_secs = wait_api_call_secs
+        self.max_wait_time = max_wait_time
+
+    def get_cube_base_url(self, subdomain: str, url: str) -> str:
+        """
+        Get Cube.js base URL.
+
+        Args:
+            - subdomain (str): Cube Cloud subdomain.
+            - url (str): Cube custom URL
+                (likely to be used in Cube.js self-hosted deployments).
+
+        Returns:
+            - Cube.js API base url.
+        """
+        cube_base_url = self.__CUBEJS_CLOUD_BASE_URL
+        if subdomain:
+            cube_base_url = f"{cube_base_url.format(subdomain=subdomain)}/cubejs-api"
+        else:
+            cube_base_url = url
+        return cube_base_url
+
+    def get_query_api_url(self) -> str:
+        """
+        Get Cube.js Query API URL.
+
+        Returns:
+            - Cube.js Query API URL.
+        """
+        return f"{self.cube_base_url}/v1/load"
+
+    def get_generated_sql_api_url(self) -> str:
+        """
+        Get Cube.js Query SQL API URL.
+
+        Returns:
+            - Cube.js Query SQL API URL.
+        """
+
+        return f"{self.cube_base_url}/v1/sql"
+
+    def get_api_token(self, security_context: Union[str, Dict], secret: str) -> str:
```

Comment:
I think you should use `self.security_context` and `self.secret` instead of passing those values in.
```suggestion
    def _get_api_token(self) -> str:
```